base/strings/string.jl

```diff
@@ -29,16 +29,28 @@ const ByteArray = Union{CodeUnits{UInt8,String}, Vector{UInt8},Vector{Int8}, Fas
 
 @inline between(b::T, lo::T, hi::T) where {T<:Integer} = (lo ≤ b) & (b ≤ hi)
 
+"""
+    String <: AbstractString
+
+The generic, default string type in Julia, used by e.g. string literals.
+
+`String`s are immutable sequences of `Char`s. A `String` is stored internally as
+a contiguous byte array, and while they are interpreted as being UTF-8 encoded,
+they can be composed of any byte sequence.
+This representation often makes `String` appropriate for passing strings to C.
+"""
+String
+
 ## constructors and conversions ##
 
 # String constructor docstring from boot.jl, workaround for #16730
 # and the unavailability of @doc in boot.jl context.
 """
     String(v::AbstractVector{UInt8})
 
-Create a new `String` object from a byte vector `v` containing UTF-8 encoded
-characters. If `v` is `Vector{UInt8}` it will be truncated to zero length and
-future modification of `v` cannot affect the contents of the resulting string.
+Create a new `String` object using the data buffer from byte vector `v`.
+If `v` is `Vector{UInt8}` it will be truncated to zero length and future
```

Comment:
Typo or intentional? 
"... `v` is `Vector{`..." 
instead of 
"... `v` is a `Vector{`..."

Having been bitten by this behaviour before I'd tend to use more drastic words, like "Create a new `String` object by depleting `v` (i.e. `v` will be empty afterwards)..."

Reply:
I agree we better put `a` there

```suggestion
If `v` is a `Vector{UInt8}` it will be truncated to zero length and future
```

(I get the second point but I'm neutral about the change)

Reply:
I think it's clear enough that the input vector is truncated.